app/src/main/java/com/github/geohunt/app/authentication/Authenticator.java

```diff
@@ -0,0 +1,17 @@
+package com.github.geohunt.app.authentication;
+
+import android.content.Context;
+
+import androidx.activity.ComponentActivity;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+public interface Authenticator<UserType> {
+    /**
+     * Invites the user to authenticate by creating an intent.
+     */
+    CompletableFuture<Optional<UserType>> authenticate(ComponentActivity activity);

```

Comment:
Instead of returning an `CompletableFuture<Optional<FirebaseUser>>` I think you should just return an `CompletableFuture<FirebaseUser>`. It will make the handling process much easier than having to check for the optional each time we log in.

Reply:
You are definitely right, I totally forgot why I did not make this at first.
Thanks!